model-optimizer/extensions/middle/ReplaceSpliceNodePattern.py

```diff
@@ -105,8 +108,11 @@ def replace_pattern(graph: Graph, match: dict):
 
             # create separate splice construction for const_dim
             memory_pair_id = unique_id('memory_for_const_dim')
-            init_value_input_memory_const_dim = create_zero_value_with_batch_from_input(split.out_port(1),
-                                                                                        memory_size_constdim)
+            init_value_input_memory_const_dim = Const(graph, {'name': 'init_value_const_dim_in_memory',
+                                                              'value': np.zeros(int64_array([in_shape[0],
```

Comment:
how does it help reshaping such networks?

Reply:
Remove shapeof construction from this transformation because it will be created a result of separate transformation. So adding such constructions here is just not required.

Reply:
Even though the ShapeOf subgraph will be created as part of another transformation why we need this change? I mean that we probably should modify the "separate transformation" so it will not execute for ReadValue operations with already properly created inputs. Since we already have working implementation with of `create_zero_value_with_batch_from_input` why we should remove it?!

Reply:
It is possible to save create_const_value_with_batch_from_input here but in this case we will have a lot of shapeof subgraphs in different places of graph. Adding batch for all ReadValue init values in one transformation allows to have one shapeof subgraph.